### PR TITLE
Release/v6.6.1

### PR DESCRIPTION
**Pull request by kttkjl**
IPR Client v6.6.1 Release Notes

Bugfixes:
- [DEVSU-1683] - Fixed CivicCellRenderer to take into account when externalSource is of type array
- [DEVSU-1686] - Added tpm and kIQR for SM, CNV, SV

### CODE REVIEW

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 1]
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
...
 
 type CivicCellRendererProps = ICellRendererParams['data'];
 
 const CivicCellRenderer = ({
   data,
 }: CivicCellRendererProps): JSX.Element => {
+  const {
+    externalStatementId,
+    externalSource,
+  } = data;
```
Comment (Anna-log7) [inline app/components/DataTable/components/CivicCellRenderer/index.tsx:17]: You can move this destructure into the props like:

```
const CivicCellRenderer = ({
  data: { externalStatementId, externalSource },
}: CivicCellRendererProps): JSX.Element => {
```

**Review by Anna-log7 — CHANGES_REQUESTED** [review]

File: app/components/DataTable/components/CivicCellRenderer/index.tsx (hunk 1 above)
Comment (kttkjl) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:17]: I could, but this I feel like it's cleaner?

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 2]
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
...
 
   useEffect(() => {
-    if (data?.externalSource?.toLowerCase() === 'civic'
-      && !Number.isNaN(parseInt(data.externalStatementId, 10))) {
-      setLink(`https://civicdb.org/links/evidence/${parseInt(data.externalStatementId, 10)}`);
-      setText(data.externalSource);
-    } else {
-      setText(data.externalSource);
+    if (data) {
+      if (Array.isArray(externalSource)) {
```
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:29]: good catch, thanks!

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 3]
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
...
+        const numOnly = /^\d+$/;
+        if (externalSource.map((es) => es.toLowerCase()).includes('civic')) {
```
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:31]: I thought externalSource was of type string, when would it be boolean?

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 4]
```diff
@@ -3,33 +3,88 @@ import React, { useState, useEffect } from 'react';
...
+  const menuItems = links.map((linkId) => (
+    <MenuItem
+      key={linkId}
+      onClick={handleMenuClose}
+    >
+      <NewTabLink link={`https://civicdb.org/links/evidence/${linkId}`} text={linkId} />
+    </MenuItem>
+  ));
+
+  if (links.length > 1) {
```
[the remark this answers is not in the document]
Comment (kttkjl) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:66]: meaning we'll have to show their IDs too? 
Previously it was just setting the text of the external sources in the cell

there are stories for these situations now on storybook

File: app/components/DataTable/components/CivicCellRenderer/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:31]: Argh I read this wrong sorry. The parentheses got me

File: app/components/DataTable/components/CivicCellRenderer/index.tsx (hunk 1 above)
Comment (Anna-log7) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:17]: Up to you then if you feel that way

File: app/components/DataTable/components/CivicCellRenderer/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:66]: Nope, you've covered all the cases it looks like!

**Comment by Anna-log7** [comment]
Only other change I'd say is to make the links the same sort of style (both blue with underline). I've been only using icons in the actions column and I think it makes sense to stick with that for now.

image

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 5]
```diff
@@ -3,33 +3,94 @@ import React, { useState, useEffect } from 'react';
...
 const CivicCellRenderer = ({
   data,
 }: CivicCellRendererProps): JSX.Element => {
+  const {
+    externalStatementId,
+    externalSource,
+  } = data;
+
   const [link, setLink] = useState('');
+  const [links, setLinks] = useState([]);
```
Comment (Anna-log7) [inline app/components/DataTable/components/CivicCellRenderer/index.tsx:20]: Should add a type here

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 6]
```diff
@@ -3,33 +3,94 @@ import React, { useState, useEffect } from 'react';
...
   const [text, setText] = useState('');
+  const [anchorEl, setAnchorEl] = useState(null);
```
Comment (Anna-log7) [inline app/components/DataTable/components/CivicCellRenderer/index.tsx:22]: Same for here with the type. You can type the event too in the handler below

**Comment by kttkjl** [comment]
> Only other change I'd say is to make the links the same sort of style (both blue with underline). I've been only using icons in the actions column and I think it makes sense to stick with that for now.
> 
> image

image

The top is when civic is array
The bottom is when civic is just a single value

**Review by Anna-log7 — DISMISSED** [review]

**Review by elewis2 — DISMISSED** [review]

**Review by Anna-log7 — APPROVED** [review]

**Pull request merged**